src/prefect/blocks/core.py

```diff
@@ -272,6 +273,15 @@ class Block(BaseModel, ABC):
     )
 
     def __init__(self, *args, **kwargs):
+        block_document_id = kwargs.pop("$ref", None)
+        if block_document_id:
+            block_document, block_document_name = self._get_block_document_by_id(
```

Comment:
because this is _always_ run synchronously, I suggest adding `_sync=True` to this call to ensure it's called properly.